{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8ee6a06f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3244df0",
   "metadata": {},
   "source": [
    "Load dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0459749a",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"/home/arog/Documents/GitHub/HGMB_Project/datasets/GutFeelingKnowledgeBase-v4-Master_List.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2ba0ee02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unique ID</th>\n",
       "      <th>UP ID</th>\n",
       "      <th>Present in GFKB v3 (Y/N)</th>\n",
       "      <th>Present in GFKB_epilepsy v3 (Y/N)</th>\n",
       "      <th>UP taxonomy</th>\n",
       "      <th>UP name</th>\n",
       "      <th>Assembly ID (UP matched)</th>\n",
       "      <th>Genome Size (Mb)</th>\n",
       "      <th>RefSeq Accession</th>\n",
       "      <th>Plasmid/additional chromosome accessions</th>\n",
       "      <th>GenBank Accession</th>\n",
       "      <th>GenBank record name</th>\n",
       "      <th>NCBI taxonomy</th>\n",
       "      <th>NCBI taxonomy name</th>\n",
       "      <th>Lineage (full)</th>\n",
       "      <th>Characteristics</th>\n",
       "      <th>Effect/Function</th>\n",
       "      <th>Curation Notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UP000001410_AE014075</td>\n",
       "      <td>UP000001410</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>199310</td>\n",
       "      <td>Escherichia coli O6:H1 (strain CFT073 / ATCC 7...</td>\n",
       "      <td>GCA_000007445.1</td>\n",
       "      <td>5.231</td>\n",
       "      <td>NC_004431.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AE014075</td>\n",
       "      <td>Escherichia coli CFT073, complete genome</td>\n",
       "      <td>199310.0</td>\n",
       "      <td>Escherichia coli CFT073</td>\n",
       "      <td>Bacteria; Proteobacteria; Gammaproteobacteria;...</td>\n",
       "      <td>Escherichia coli is a Gram-negative straight r...</td>\n",
       "      <td>Urinary tract infections (UTIs) are among the ...</td>\n",
       "      <td>Escherichia coli Nissle 1917, complete genome ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UP000000439_AE014295</td>\n",
       "      <td>UP000000439</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>206672</td>\n",
       "      <td>Bifidobacterium longum (strain NCC 2705)</td>\n",
       "      <td>GCA_000007525.1</td>\n",
       "      <td>2.260</td>\n",
       "      <td>NC_004307</td>\n",
       "      <td>NC_004943.1</td>\n",
       "      <td>AE014295</td>\n",
       "      <td>Bifidobacterium longum NCC2705 complete genome</td>\n",
       "      <td>206672.0</td>\n",
       "      <td>Bifidobacterium longum NCC2705</td>\n",
       "      <td>cellular organisms; Bacteria; Terrabacteria gr...</td>\n",
       "      <td>Bifidobacterium longum is an anaerobic, non-ha...</td>\n",
       "      <td>The ability of Bifidobacterium spp. to survive...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UP000001414_AE015928</td>\n",
       "      <td>UP000001414</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>226186</td>\n",
       "      <td>Bacteroides thetaiotaomicron (strain ATCC 2914...</td>\n",
       "      <td>GCA_000011065.1</td>\n",
       "      <td>6.293</td>\n",
       "      <td>NC_004663.1</td>\n",
       "      <td>NC_004703.1</td>\n",
       "      <td>AE015928</td>\n",
       "      <td>Bacteroides thetaiotaomicron VPI-5482, complet...</td>\n",
       "      <td>226186.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron VPI-5482</td>\n",
       "      <td>Bacteria; FCB group; Bacteroidetes/Chlorobi gr...</td>\n",
       "      <td>The human distal gut microbiota contain more b...</td>\n",
       "      <td>This organism has many types of mobile genetic...</td>\n",
       "      <td>Bacteroides thetaiotaomicron VPI-5482 plasmid ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UP000001978_AM180355.1</td>\n",
       "      <td>UP000001978</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>272563</td>\n",
       "      <td>Peptoclostridium difficile (strain 630)</td>\n",
       "      <td>GCA_000009205.2</td>\n",
       "      <td>4.298</td>\n",
       "      <td>NC_009089.1</td>\n",
       "      <td>NC_008226.2</td>\n",
       "      <td>AM180355.1</td>\n",
       "      <td>Clostridium difficile 630 complete genome</td>\n",
       "      <td>272563.0</td>\n",
       "      <td>Clostridioides difficile 630</td>\n",
       "      <td>Bacteria; Terrabacteria group; Firmicutes; Clo...</td>\n",
       "      <td>Clostridium difficile is a Gram-positive, anae...</td>\n",
       "      <td>This multiresistant strain that was isolated f...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>UP000002197_AP006841</td>\n",
       "      <td>UP000002197</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>295405</td>\n",
       "      <td>Bacteroides fragilis (strain YCH46)</td>\n",
       "      <td>GCA_000009925.1</td>\n",
       "      <td>5.311</td>\n",
       "      <td>NC_006347.1</td>\n",
       "      <td>NC_006297.1</td>\n",
       "      <td>AP006841</td>\n",
       "      <td>Bacteroides fragilis YCH46 DNA, complete genome</td>\n",
       "      <td>295405.0</td>\n",
       "      <td>Bacteroides fragilis YCH46</td>\n",
       "      <td>Bacteria; FCB group; Bacteroidetes/Chlorobi gr...</td>\n",
       "      <td>Bacteroides fragilis is a Gram-negative rod, w...</td>\n",
       "      <td>This strain contains many moble genetic elemen...</td>\n",
       "      <td>Bacteroides fragilis YCH46 plasmid pBFY46 DNA,...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Unique ID        UP ID Present in GFKB v3 (Y/N)  \\\n",
       "0    UP000001410_AE014075  UP000001410                        Y   \n",
       "1    UP000000439_AE014295  UP000000439                        N   \n",
       "2    UP000001414_AE015928  UP000001414                        Y   \n",
       "3  UP000001978_AM180355.1  UP000001978                        Y   \n",
       "4    UP000002197_AP006841  UP000002197                        Y   \n",
       "\n",
       "  Present in GFKB_epilepsy v3 (Y/N)  UP taxonomy  \\\n",
       "0                                 Y       199310   \n",
       "1                                 Y       206672   \n",
       "2                                 Y       226186   \n",
       "3                                 N       272563   \n",
       "4                                 Y       295405   \n",
       "\n",
       "                                             UP name Assembly ID (UP matched)  \\\n",
       "0  Escherichia coli O6:H1 (strain CFT073 / ATCC 7...          GCA_000007445.1   \n",
       "1           Bifidobacterium longum (strain NCC 2705)          GCA_000007525.1   \n",
       "2  Bacteroides thetaiotaomicron (strain ATCC 2914...          GCA_000011065.1   \n",
       "3            Peptoclostridium difficile (strain 630)          GCA_000009205.2   \n",
       "4                Bacteroides fragilis (strain YCH46)          GCA_000009925.1   \n",
       "\n",
       "   Genome Size (Mb) RefSeq Accession Plasmid/additional chromosome accessions  \\\n",
       "0             5.231      NC_004431.1                                      NaN   \n",
       "1             2.260        NC_004307                              NC_004943.1   \n",
       "2             6.293      NC_004663.1                              NC_004703.1   \n",
       "3             4.298      NC_009089.1                              NC_008226.2   \n",
       "4             5.311      NC_006347.1                              NC_006297.1   \n",
       "\n",
       "  GenBank Accession                                GenBank record name  \\\n",
       "0          AE014075           Escherichia coli CFT073, complete genome   \n",
       "1          AE014295     Bifidobacterium longum NCC2705 complete genome   \n",
       "2          AE015928  Bacteroides thetaiotaomicron VPI-5482, complet...   \n",
       "3        AM180355.1          Clostridium difficile 630 complete genome   \n",
       "4          AP006841    Bacteroides fragilis YCH46 DNA, complete genome   \n",
       "\n",
       "   NCBI taxonomy                     NCBI taxonomy name  \\\n",
       "0       199310.0                Escherichia coli CFT073   \n",
       "1       206672.0         Bifidobacterium longum NCC2705   \n",
       "2       226186.0  Bacteroides thetaiotaomicron VPI-5482   \n",
       "3       272563.0           Clostridioides difficile 630   \n",
       "4       295405.0             Bacteroides fragilis YCH46   \n",
       "\n",
       "                                      Lineage (full)  \\\n",
       "0  Bacteria; Proteobacteria; Gammaproteobacteria;...   \n",
       "1  cellular organisms; Bacteria; Terrabacteria gr...   \n",
       "2  Bacteria; FCB group; Bacteroidetes/Chlorobi gr...   \n",
       "3  Bacteria; Terrabacteria group; Firmicutes; Clo...   \n",
       "4  Bacteria; FCB group; Bacteroidetes/Chlorobi gr...   \n",
       "\n",
       "                                     Characteristics  \\\n",
       "0  Escherichia coli is a Gram-negative straight r...   \n",
       "1  Bifidobacterium longum is an anaerobic, non-ha...   \n",
       "2  The human distal gut microbiota contain more b...   \n",
       "3  Clostridium difficile is a Gram-positive, anae...   \n",
       "4  Bacteroides fragilis is a Gram-negative rod, w...   \n",
       "\n",
       "                                     Effect/Function  \\\n",
       "0  Urinary tract infections (UTIs) are among the ...   \n",
       "1  The ability of Bifidobacterium spp. to survive...   \n",
       "2  This organism has many types of mobile genetic...   \n",
       "3  This multiresistant strain that was isolated f...   \n",
       "4  This strain contains many moble genetic elemen...   \n",
       "\n",
       "                                      Curation Notes  \n",
       "0  Escherichia coli Nissle 1917, complete genome ...  \n",
       "1                                                NaN  \n",
       "2  Bacteroides thetaiotaomicron VPI-5482 plasmid ...  \n",
       "3                                                NaN  \n",
       "4  Bacteroides fragilis YCH46 plasmid pBFY46 DNA,...  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83b820bb",
   "metadata": {},
   "source": [
    "Prepare dataset for crossref with only healthy gut specimens"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b4a84706",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter rows where \"Present in GFKB v3 (Y/N)\" is \"Y\"\n",
    "filtered_df = df[df['Present in GFKB v3 (Y/N)'] == 'Y'].copy()\n",
    "\n",
    "# Select only \"NCBI taxonomy name\" and add new columns\n",
    "result_df = filtered_df[['NCBI taxonomy name']].copy()\n",
    "\n",
    "# Add new columns (default 'N' and 'NA')\n",
    "result_df['In BiGG Models (Y/N)'] = 'N'  # Default 'N' (can be updated later)\n",
    "result_df['BiGG Model Link'] = 'NA'      # Default 'NA' (can be updated later)\n",
    "\n",
    "result_df.to_csv('/home/arog/Documents/GitHub/HGMB_Project/datasets/treated/crossref_GFKB3.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "geral",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
