{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8ee6a06f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3244df0",
   "metadata": {},
   "source": [
    "### Load dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0459749a",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_v3 = pd.read_csv(\"/home/arog/Documents/GitHub/HGMB_Project/datasets/GFKB/GutFeelingKnowledgeBase-v4-Master_List.csv\")\n",
    "df_v3_abund = pd.read_csv(\n",
    "    \"/home/arog/Documents/GitHub/HGMB_Project/datasets/GFKB/GutFeelingKnowledgeBase-v4-Epilepsy_Data.csv\",\n",
    "    skiprows=1\n",
    ")\n",
    "df_v5 = pd.read_csv(\"/home/arog/Documents/GitHub/HGMB_Project/datasets/GFKB/GFKB_v5_PreDiabetes.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2ba0ee02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unique ID</th>\n",
       "      <th>UP ID</th>\n",
       "      <th>Present in GFKB v3 (Y/N)</th>\n",
       "      <th>Present in GFKB_epilepsy v3 (Y/N)</th>\n",
       "      <th>UP taxonomy</th>\n",
       "      <th>UP name</th>\n",
       "      <th>Assembly ID (UP matched)</th>\n",
       "      <th>Genome Size (Mb)</th>\n",
       "      <th>RefSeq Accession</th>\n",
       "      <th>Plasmid/additional chromosome accessions</th>\n",
       "      <th>GenBank Accession</th>\n",
       "      <th>GenBank record name</th>\n",
       "      <th>NCBI taxonomy</th>\n",
       "      <th>NCBI taxonomy name</th>\n",
       "      <th>Lineage (full)</th>\n",
       "      <th>Characteristics</th>\n",
       "      <th>Effect/Function</th>\n",
       "      <th>Curation Notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UP000001410_AE014075</td>\n",
       "      <td>UP000001410</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>199310</td>\n",
       "      <td>Escherichia coli O6:H1 (strain CFT073 / ATCC 7...</td>\n",
       "      <td>GCA_000007445.1</td>\n",
       "      <td>5.231</td>\n",
       "      <td>NC_004431.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AE014075</td>\n",
       "      <td>Escherichia coli CFT073, complete genome</td>\n",
       "      <td>199310.0</td>\n",
       "      <td>Escherichia coli CFT073</td>\n",
       "      <td>Bacteria; Proteobacteria; Gammaproteobacteria;...</td>\n",
       "      <td>Escherichia coli is a Gram-negative straight r...</td>\n",
       "      <td>Urinary tract infections (UTIs) are among the ...</td>\n",
       "      <td>Escherichia coli Nissle 1917, complete genome ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UP000000439_AE014295</td>\n",
       "      <td>UP000000439</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>206672</td>\n",
       "      <td>Bifidobacterium longum (strain NCC 2705)</td>\n",
       "      <td>GCA_000007525.1</td>\n",
       "      <td>2.260</td>\n",
       "      <td>NC_004307</td>\n",
       "      <td>NC_004943.1</td>\n",
       "      <td>AE014295</td>\n",
       "      <td>Bifidobacterium longum NCC2705 complete genome</td>\n",
       "      <td>206672.0</td>\n",
       "      <td>Bifidobacterium longum NCC2705</td>\n",
       "      <td>cellular organisms; Bacteria; Terrabacteria gr...</td>\n",
       "      <td>Bifidobacterium longum is an anaerobic, non-ha...</td>\n",
       "      <td>The ability of Bifidobacterium spp. to survive...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UP000001414_AE015928</td>\n",
       "      <td>UP000001414</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>226186</td>\n",
       "      <td>Bacteroides thetaiotaomicron (strain ATCC 2914...</td>\n",
       "      <td>GCA_000011065.1</td>\n",
       "      <td>6.293</td>\n",
       "      <td>NC_004663.1</td>\n",
       "      <td>NC_004703.1</td>\n",
       "      <td>AE015928</td>\n",
       "      <td>Bacteroides thetaiotaomicron VPI-5482, complet...</td>\n",
       "      <td>226186.0</td>\n",
       "      <td>Bacteroides thetaiotaomicron VPI-5482</td>\n",
       "      <td>Bacteria; FCB group; Bacteroidetes/Chlorobi gr...</td>\n",
       "      <td>The human distal gut microbiota contain more b...</td>\n",
       "      <td>This organism has many types of mobile genetic...</td>\n",
       "      <td>Bacteroides thetaiotaomicron VPI-5482 plasmid ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UP000001978_AM180355.1</td>\n",
       "      <td>UP000001978</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>272563</td>\n",
       "      <td>Peptoclostridium difficile (strain 630)</td>\n",
       "      <td>GCA_000009205.2</td>\n",
       "      <td>4.298</td>\n",
       "      <td>NC_009089.1</td>\n",
       "      <td>NC_008226.2</td>\n",
       "      <td>AM180355.1</td>\n",
       "      <td>Clostridium difficile 630 complete genome</td>\n",
       "      <td>272563.0</td>\n",
       "      <td>Clostridioides difficile 630</td>\n",
       "      <td>Bacteria; Terrabacteria group; Firmicutes; Clo...</td>\n",
       "      <td>Clostridium difficile is a Gram-positive, anae...</td>\n",
       "      <td>This multiresistant strain that was isolated f...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>UP000002197_AP006841</td>\n",
       "      <td>UP000002197</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>295405</td>\n",
       "      <td>Bacteroides fragilis (strain YCH46)</td>\n",
       "      <td>GCA_000009925.1</td>\n",
       "      <td>5.311</td>\n",
       "      <td>NC_006347.1</td>\n",
       "      <td>NC_006297.1</td>\n",
       "      <td>AP006841</td>\n",
       "      <td>Bacteroides fragilis YCH46 DNA, complete genome</td>\n",
       "      <td>295405.0</td>\n",
       "      <td>Bacteroides fragilis YCH46</td>\n",
       "      <td>Bacteria; FCB group; Bacteroidetes/Chlorobi gr...</td>\n",
       "      <td>Bacteroides fragilis is a Gram-negative rod, w...</td>\n",
       "      <td>This strain contains many moble genetic elemen...</td>\n",
       "      <td>Bacteroides fragilis YCH46 plasmid pBFY46 DNA,...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Unique ID        UP ID Present in GFKB v3 (Y/N)  \\\n",
       "0    UP000001410_AE014075  UP000001410                        Y   \n",
       "1    UP000000439_AE014295  UP000000439                        N   \n",
       "2    UP000001414_AE015928  UP000001414                        Y   \n",
       "3  UP000001978_AM180355.1  UP000001978                        Y   \n",
       "4    UP000002197_AP006841  UP000002197                        Y   \n",
       "\n",
       "  Present in GFKB_epilepsy v3 (Y/N)  UP taxonomy  \\\n",
       "0                                 Y       199310   \n",
       "1                                 Y       206672   \n",
       "2                                 Y       226186   \n",
       "3                                 N       272563   \n",
       "4                                 Y       295405   \n",
       "\n",
       "                                             UP name Assembly ID (UP matched)  \\\n",
       "0  Escherichia coli O6:H1 (strain CFT073 / ATCC 7...          GCA_000007445.1   \n",
       "1           Bifidobacterium longum (strain NCC 2705)          GCA_000007525.1   \n",
       "2  Bacteroides thetaiotaomicron (strain ATCC 2914...          GCA_000011065.1   \n",
       "3            Peptoclostridium difficile (strain 630)          GCA_000009205.2   \n",
       "4                Bacteroides fragilis (strain YCH46)          GCA_000009925.1   \n",
       "\n",
       "   Genome Size (Mb) RefSeq Accession Plasmid/additional chromosome accessions  \\\n",
       "0             5.231      NC_004431.1                                      NaN   \n",
       "1             2.260        NC_004307                              NC_004943.1   \n",
       "2             6.293      NC_004663.1                              NC_004703.1   \n",
       "3             4.298      NC_009089.1                              NC_008226.2   \n",
       "4             5.311      NC_006347.1                              NC_006297.1   \n",
       "\n",
       "  GenBank Accession                                GenBank record name  \\\n",
       "0          AE014075           Escherichia coli CFT073, complete genome   \n",
       "1          AE014295     Bifidobacterium longum NCC2705 complete genome   \n",
       "2          AE015928  Bacteroides thetaiotaomicron VPI-5482, complet...   \n",
       "3        AM180355.1          Clostridium difficile 630 complete genome   \n",
       "4          AP006841    Bacteroides fragilis YCH46 DNA, complete genome   \n",
       "\n",
       "   NCBI taxonomy                     NCBI taxonomy name  \\\n",
       "0       199310.0                Escherichia coli CFT073   \n",
       "1       206672.0         Bifidobacterium longum NCC2705   \n",
       "2       226186.0  Bacteroides thetaiotaomicron VPI-5482   \n",
       "3       272563.0           Clostridioides difficile 630   \n",
       "4       295405.0             Bacteroides fragilis YCH46   \n",
       "\n",
       "                                      Lineage (full)  \\\n",
       "0  Bacteria; Proteobacteria; Gammaproteobacteria;...   \n",
       "1  cellular organisms; Bacteria; Terrabacteria gr...   \n",
       "2  Bacteria; FCB group; Bacteroidetes/Chlorobi gr...   \n",
       "3  Bacteria; Terrabacteria group; Firmicutes; Clo...   \n",
       "4  Bacteria; FCB group; Bacteroidetes/Chlorobi gr...   \n",
       "\n",
       "                                     Characteristics  \\\n",
       "0  Escherichia coli is a Gram-negative straight r...   \n",
       "1  Bifidobacterium longum is an anaerobic, non-ha...   \n",
       "2  The human distal gut microbiota contain more b...   \n",
       "3  Clostridium difficile is a Gram-positive, anae...   \n",
       "4  Bacteroides fragilis is a Gram-negative rod, w...   \n",
       "\n",
       "                                     Effect/Function  \\\n",
       "0  Urinary tract infections (UTIs) are among the ...   \n",
       "1  The ability of Bifidobacterium spp. to survive...   \n",
       "2  This organism has many types of mobile genetic...   \n",
       "3  This multiresistant strain that was isolated f...   \n",
       "4  This strain contains many moble genetic elemen...   \n",
       "\n",
       "                                      Curation Notes  \n",
       "0  Escherichia coli Nissle 1917, complete genome ...  \n",
       "1                                                NaN  \n",
       "2  Bacteroides thetaiotaomicron VPI-5482 plasmid ...  \n",
       "3                                                NaN  \n",
       "4  Bacteroides fragilis YCH46 plasmid pBFY46 DNA,...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_v3.dropna(how='all', inplace=True)\n",
    "df_v3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d00e2713",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unique ID</th>\n",
       "      <th>UP ID</th>\n",
       "      <th>Present in GFKB v3 (Y/N)</th>\n",
       "      <th>Present in GFKB_epilepsy v3 (Y/N)</th>\n",
       "      <th>UP taxonomy</th>\n",
       "      <th>UP name</th>\n",
       "      <th>Assembly ID (UP matched)</th>\n",
       "      <th>Genome Size (Mb)</th>\n",
       "      <th>RefSeq Accession</th>\n",
       "      <th>Plasmid/additional chromosome accessions</th>\n",
       "      <th>GenBank Accession</th>\n",
       "      <th>GenBank record name</th>\n",
       "      <th>NCBI taxonomy</th>\n",
       "      <th>NCBI taxonomy name</th>\n",
       "      <th>Lineage (full)</th>\n",
       "      <th>Characteristics</th>\n",
       "      <th>Effect/Function</th>\n",
       "      <th>Curation Notes</th>\n",
       "      <th>Additional Notes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UP000318946_AP019735</td>\n",
       "      <td>UP000318946</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>2585118.0</td>\n",
       "      <td>Alistipes communis strain 5CBH24</td>\n",
       "      <td>GCA_006542665.1</td>\n",
       "      <td>3.3</td>\n",
       "      <td>NZ_AP019735.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP019735</td>\n",
       "      <td>Alistipes communis strain 5CBH24 chromosome, c...</td>\n",
       "      <td>2585118.0</td>\n",
       "      <td>Alistipes communis 5CBH24</td>\n",
       "      <td>cellular organisms; Bacteria; FCB group; Bacte...</td>\n",
       "      <td>This strain was isolated from the gut microbio...</td>\n",
       "      <td>The Alistipes genus appears to have protective...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UP000317465_AP019737</td>\n",
       "      <td>UP000317465</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>2585117.0</td>\n",
       "      <td>Alistipes onderdonkii subsp. vulgaris strain 5...</td>\n",
       "      <td>GCA_006542705.1</td>\n",
       "      <td>3.3</td>\n",
       "      <td>NZ_AP019737.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AP019737</td>\n",
       "      <td>Alistipes onderdonkii subsp. vulgaris 5CPYCFAH...</td>\n",
       "      <td>2585117.0</td>\n",
       "      <td>Alistipes onderdonkii subsp. vulgaris strain 5...</td>\n",
       "      <td>cellular organisms; Bacteria; FCB group; Bacte...</td>\n",
       "      <td>This strain belongs to the subspecies vulgaris...</td>\n",
       "      <td>A study found A. onderdonkii medium inhibited ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UP000237636_CP027234</td>\n",
       "      <td>UP000237636</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>28113.0</td>\n",
       "      <td>Bacteroides heparinolyticus strain F0111</td>\n",
       "      <td>GCA_002998535.1</td>\n",
       "      <td>3.6</td>\n",
       "      <td>NZ_CP027234.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CP027234</td>\n",
       "      <td>Bacteroides heparinolyticus strain F0111 chrom...</td>\n",
       "      <td>28113.0</td>\n",
       "      <td>Bacteroides heparinolyticus strain F0111</td>\n",
       "      <td>cellular organisms; Bacteria; FCB group; Bacte...</td>\n",
       "      <td>This strain is a gram-negative, anaerobic, rod...</td>\n",
       "      <td>A study indicates B.heparinolyticus produces a...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteroides heparinolyticus is synonymous with...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UP000294024_CP036491</td>\n",
       "      <td>UP000294024</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>2292922.0</td>\n",
       "      <td>Bacteroides sp. A1C1</td>\n",
       "      <td>GCA_004328515.1</td>\n",
       "      <td>4.8</td>\n",
       "      <td>NZ_CP036491.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CP036491</td>\n",
       "      <td>Bacteroides sp. A1C1 chromosome, complete genome</td>\n",
       "      <td>2292922.0</td>\n",
       "      <td>Bacteroides sp. A1C1</td>\n",
       "      <td>cellular organisms; Bacteria; FCB group; Bacte...</td>\n",
       "      <td>This strain is a gram-negative rod that is non...</td>\n",
       "      <td>A study found Bacteroides A1C1 and Bacteroides...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This strain has been found in cat and chincill...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>UP000316618_VLLM00000000</td>\n",
       "      <td>UP000316618</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>28119.0</td>\n",
       "      <td>Bacteroides zoogleoformans strain ATCC 33285</td>\n",
       "      <td>GCA_007830715.1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLLM00000000</td>\n",
       "      <td>Bacteroides zoogleoformans strain ATCC 33285, ...</td>\n",
       "      <td>28119.0</td>\n",
       "      <td>Bacteroides zoogleoformans strain ATCC 33285</td>\n",
       "      <td>cellular organisms; Bacteria; FCB group; Bacte...</td>\n",
       "      <td>Bacteroides zooleoformans are anaerobic, non-s...</td>\n",
       "      <td>They function to ferment carbohydrates and pro...</td>\n",
       "      <td>This accession is a WGS project, and contains ...</td>\n",
       "      <td>Bacteroides zoogleoformans is synonymous with ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Unique ID        UP ID Present in GFKB v3 (Y/N)  \\\n",
       "0      UP000318946_AP019735  UP000318946                        N   \n",
       "1      UP000317465_AP019737  UP000317465                        N   \n",
       "2     UP000237636_CP027234   UP000237636                        N   \n",
       "3      UP000294024_CP036491  UP000294024                        N   \n",
       "4  UP000316618_VLLM00000000  UP000316618                        N   \n",
       "\n",
       "  Present in GFKB_epilepsy v3 (Y/N)  UP taxonomy  \\\n",
       "0                                 N    2585118.0   \n",
       "1                                 N    2585117.0   \n",
       "2                                 N      28113.0   \n",
       "3                                 N    2292922.0   \n",
       "4                                 N      28119.0   \n",
       "\n",
       "                                             UP name Assembly ID (UP matched)  \\\n",
       "0                   Alistipes communis strain 5CBH24          GCA_006542665.1   \n",
       "1  Alistipes onderdonkii subsp. vulgaris strain 5...          GCA_006542705.1   \n",
       "2           Bacteroides heparinolyticus strain F0111          GCA_002998535.1   \n",
       "3                               Bacteroides sp. A1C1          GCA_004328515.1   \n",
       "4       Bacteroides zoogleoformans strain ATCC 33285          GCA_007830715.1   \n",
       "\n",
       "   Genome Size (Mb) RefSeq Accession Plasmid/additional chromosome accessions  \\\n",
       "0               3.3    NZ_AP019735.1                                      NaN   \n",
       "1               3.3    NZ_AP019737.1                                      NaN   \n",
       "2               3.6    NZ_CP027234.1                                      NaN   \n",
       "3               4.8    NZ_CP036491.1                                      NaN   \n",
       "4               1.0              NaN                                      NaN   \n",
       "\n",
       "  GenBank Accession                                GenBank record name  \\\n",
       "0          AP019735  Alistipes communis strain 5CBH24 chromosome, c...   \n",
       "1          AP019737  Alistipes onderdonkii subsp. vulgaris 5CPYCFAH...   \n",
       "2          CP027234  Bacteroides heparinolyticus strain F0111 chrom...   \n",
       "3          CP036491   Bacteroides sp. A1C1 chromosome, complete genome   \n",
       "4      VLLM00000000  Bacteroides zoogleoformans strain ATCC 33285, ...   \n",
       "\n",
       "   NCBI taxonomy                                 NCBI taxonomy name  \\\n",
       "0      2585118.0                          Alistipes communis 5CBH24   \n",
       "1      2585117.0  Alistipes onderdonkii subsp. vulgaris strain 5...   \n",
       "2        28113.0           Bacteroides heparinolyticus strain F0111   \n",
       "3      2292922.0                               Bacteroides sp. A1C1   \n",
       "4        28119.0       Bacteroides zoogleoformans strain ATCC 33285   \n",
       "\n",
       "                                      Lineage (full)  \\\n",
       "0  cellular organisms; Bacteria; FCB group; Bacte...   \n",
       "1  cellular organisms; Bacteria; FCB group; Bacte...   \n",
       "2  cellular organisms; Bacteria; FCB group; Bacte...   \n",
       "3  cellular organisms; Bacteria; FCB group; Bacte...   \n",
       "4  cellular organisms; Bacteria; FCB group; Bacte...   \n",
       "\n",
       "                                     Characteristics  \\\n",
       "0  This strain was isolated from the gut microbio...   \n",
       "1  This strain belongs to the subspecies vulgaris...   \n",
       "2  This strain is a gram-negative, anaerobic, rod...   \n",
       "3  This strain is a gram-negative rod that is non...   \n",
       "4  Bacteroides zooleoformans are anaerobic, non-s...   \n",
       "\n",
       "                                     Effect/Function  \\\n",
       "0  The Alistipes genus appears to have protective...   \n",
       "1  A study found A. onderdonkii medium inhibited ...   \n",
       "2  A study indicates B.heparinolyticus produces a...   \n",
       "3  A study found Bacteroides A1C1 and Bacteroides...   \n",
       "4  They function to ferment carbohydrates and pro...   \n",
       "\n",
       "                                      Curation Notes  \\\n",
       "0                                                NaN   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3                                                NaN   \n",
       "4  This accession is a WGS project, and contains ...   \n",
       "\n",
       "                                    Additional Notes  \n",
       "0                                                NaN  \n",
       "1                                                NaN  \n",
       "2  Bacteroides heparinolyticus is synonymous with...  \n",
       "3  This strain has been found in cat and chincill...  \n",
       "4  Bacteroides zoogleoformans is synonymous with ...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_v5.dropna(how='all', inplace=True)\n",
    "df_v5 = df_v5.dropna(axis=1, how='all')\n",
    "df_v5.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6c83ad1f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniqueID</th>\n",
       "      <th>UP name</th>\n",
       "      <th>Lineage( full )</th>\n",
       "      <th>Median_Control_Before_Diet</th>\n",
       "      <th>Mean_Control_Before_Diet</th>\n",
       "      <th>Standard_Deviation_Control_Before_Diet</th>\n",
       "      <th>Interquartile_Range_Control_Before_Diet</th>\n",
       "      <th>Range_Control_Before_Diet</th>\n",
       "      <th>0%_Control_Before_Diet</th>\n",
       "      <th>25%_Control_Before_Diet</th>\n",
       "      <th>...</th>\n",
       "      <th>Median_Patient_After_Diet</th>\n",
       "      <th>Mean_Patient_After_Diet</th>\n",
       "      <th>Standard_Deviation_Patient_After_Diet</th>\n",
       "      <th>Interquartile_Range_Patient_After_Diet</th>\n",
       "      <th>Range_Patient_After_Diet</th>\n",
       "      <th>0%_Patient_After_Diet</th>\n",
       "      <th>25%_Patient_After_Diet</th>\n",
       "      <th>50%_Patient_After_Diet</th>\n",
       "      <th>75%_Patient_After_Diet</th>\n",
       "      <th>100%_Patient_After_Diet</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>UP000001410_AE014075</td>\n",
       "      <td>Escherichia coli O6:H1 (strain CFT073 / ATCC 7...</td>\n",
       "      <td>Bacteria; Proteobacteria; Gammaproteobacteria;...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>0.000121</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.00040</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002010</td>\n",
       "      <td>0.009343</td>\n",
       "      <td>0.016154</td>\n",
       "      <td>0.010217</td>\n",
       "      <td>0.05590</td>\n",
       "      <td>0.00017</td>\n",
       "      <td>0.000883</td>\n",
       "      <td>0.002010</td>\n",
       "      <td>0.011100</td>\n",
       "      <td>0.05607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UP000000439_AE014295</td>\n",
       "      <td>Bifidobacterium longum (strain NCC 2705)</td>\n",
       "      <td>cellular organisms; Bacteria; Terrabacteria gr...</td>\n",
       "      <td>0.00068</td>\n",
       "      <td>0.001038</td>\n",
       "      <td>0.000909</td>\n",
       "      <td>0.001460</td>\n",
       "      <td>0.00246</td>\n",
       "      <td>0.00009</td>\n",
       "      <td>0.000265</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.000423</td>\n",
       "      <td>0.000829</td>\n",
       "      <td>0.000228</td>\n",
       "      <td>0.00272</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.000262</td>\n",
       "      <td>0.00272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>UP000001414_AE015928</td>\n",
       "      <td>Bacteroides thetaiotaomicron (strain ATCC 2914...</td>\n",
       "      <td>Bacteria; FCB group; Bacteroidetes/Chlorobi gr...</td>\n",
       "      <td>0.00147</td>\n",
       "      <td>0.002365</td>\n",
       "      <td>0.002757</td>\n",
       "      <td>0.001790</td>\n",
       "      <td>0.00936</td>\n",
       "      <td>0.00042</td>\n",
       "      <td>0.000775</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001410</td>\n",
       "      <td>0.006975</td>\n",
       "      <td>0.015114</td>\n",
       "      <td>0.003903</td>\n",
       "      <td>0.05375</td>\n",
       "      <td>0.00049</td>\n",
       "      <td>0.000822</td>\n",
       "      <td>0.001410</td>\n",
       "      <td>0.004725</td>\n",
       "      <td>0.05424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>UP000001978_AM180355.1</td>\n",
       "      <td>Peptoclostridium difficile (strain 630)</td>\n",
       "      <td>Bacteria; Terrabacteria group; Firmicutes; Clo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>UP000002197_AP006841</td>\n",
       "      <td>Bacteroides fragilis (strain YCH46)</td>\n",
       "      <td>Bacteria; FCB group; Bacteroidetes/Chlorobi gr...</td>\n",
       "      <td>0.00097</td>\n",
       "      <td>0.001412</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.000910</td>\n",
       "      <td>0.00303</td>\n",
       "      <td>0.00060</td>\n",
       "      <td>0.000710</td>\n",
       "      <td>...</td>\n",
       "      <td>0.010315</td>\n",
       "      <td>0.019652</td>\n",
       "      <td>0.024551</td>\n",
       "      <td>0.027275</td>\n",
       "      <td>0.07327</td>\n",
       "      <td>0.00092</td>\n",
       "      <td>0.001538</td>\n",
       "      <td>0.010315</td>\n",
       "      <td>0.028813</td>\n",
       "      <td>0.07419</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 uniqueID                                            UP name  \\\n",
       "0    UP000001410_AE014075  Escherichia coli O6:H1 (strain CFT073 / ATCC 7...   \n",
       "1    UP000000439_AE014295           Bifidobacterium longum (strain NCC 2705)   \n",
       "2    UP000001414_AE015928  Bacteroides thetaiotaomicron (strain ATCC 2914...   \n",
       "3  UP000001978_AM180355.1            Peptoclostridium difficile (strain 630)   \n",
       "4    UP000002197_AP006841                Bacteroides fragilis (strain YCH46)   \n",
       "\n",
       "                                     Lineage( full )  \\\n",
       "0  Bacteria; Proteobacteria; Gammaproteobacteria;...   \n",
       "1  cellular organisms; Bacteria; Terrabacteria gr...   \n",
       "2  Bacteria; FCB group; Bacteroidetes/Chlorobi gr...   \n",
       "3  Bacteria; Terrabacteria group; Firmicutes; Clo...   \n",
       "4  Bacteria; FCB group; Bacteroidetes/Chlorobi gr...   \n",
       "\n",
       "   Median_Control_Before_Diet  Mean_Control_Before_Diet  \\\n",
       "0                     0.00000                  0.000056   \n",
       "1                     0.00068                  0.001038   \n",
       "2                     0.00147                  0.002365   \n",
       "3                         NaN                       NaN   \n",
       "4                     0.00097                  0.001412   \n",
       "\n",
       "   Standard_Deviation_Control_Before_Diet  \\\n",
       "0                                0.000121   \n",
       "1                                0.000909   \n",
       "2                                0.002757   \n",
       "3                                     NaN   \n",
       "4                                0.001044   \n",
       "\n",
       "   Interquartile_Range_Control_Before_Diet  Range_Control_Before_Diet  \\\n",
       "0                                 0.000035                    0.00040   \n",
       "1                                 0.001460                    0.00246   \n",
       "2                                 0.001790                    0.00936   \n",
       "3                                      NaN                        NaN   \n",
       "4                                 0.000910                    0.00303   \n",
       "\n",
       "   0%_Control_Before_Diet  25%_Control_Before_Diet  ...  \\\n",
       "0                 0.00000                 0.000000  ...   \n",
       "1                 0.00009                 0.000265  ...   \n",
       "2                 0.00042                 0.000775  ...   \n",
       "3                     NaN                      NaN  ...   \n",
       "4                 0.00060                 0.000710  ...   \n",
       "\n",
       "   Median_Patient_After_Diet  Mean_Patient_After_Diet  \\\n",
       "0                   0.002010                 0.009343   \n",
       "1                   0.000060                 0.000423   \n",
       "2                   0.001410                 0.006975   \n",
       "3                        NaN                      NaN   \n",
       "4                   0.010315                 0.019652   \n",
       "\n",
       "   Standard_Deviation_Patient_After_Diet  \\\n",
       "0                               0.016154   \n",
       "1                               0.000829   \n",
       "2                               0.015114   \n",
       "3                                    NaN   \n",
       "4                               0.024551   \n",
       "\n",
       "   Interquartile_Range_Patient_After_Diet  Range_Patient_After_Diet  \\\n",
       "0                                0.010217                   0.05590   \n",
       "1                                0.000228                   0.00272   \n",
       "2                                0.003903                   0.05375   \n",
       "3                                     NaN                       NaN   \n",
       "4                                0.027275                   0.07327   \n",
       "\n",
       "   0%_Patient_After_Diet  25%_Patient_After_Diet  50%_Patient_After_Diet  \\\n",
       "0                0.00017                0.000883                0.002010   \n",
       "1                0.00000                0.000035                0.000060   \n",
       "2                0.00049                0.000822                0.001410   \n",
       "3                    NaN                     NaN                     NaN   \n",
       "4                0.00092                0.001538                0.010315   \n",
       "\n",
       "   75%_Patient_After_Diet  100%_Patient_After_Diet  \n",
       "0                0.011100                  0.05607  \n",
       "1                0.000262                  0.00272  \n",
       "2                0.004725                  0.05424  \n",
       "3                     NaN                      NaN  \n",
       "4                0.028813                  0.07419  \n",
       "\n",
       "[5 rows x 43 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_v3_abund.dropna(how='all', inplace=True)\n",
    "df_v3_abund.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8c55ff91",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Dataset</th>\n",
       "      <th>Shape</th>\n",
       "      <th>Columns</th>\n",
       "      <th>Missing % (total)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GFKB v4 Master List</td>\n",
       "      <td>188 rows × 18 cols</td>\n",
       "      <td>Unique ID, UP ID, Present in GFKB v3 (Y/N), Pr...</td>\n",
       "      <td>6.38%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GFKB v4 Epilepsy Abund</td>\n",
       "      <td>188 rows × 43 cols</td>\n",
       "      <td>uniqueID, UP name, Lineage( full ), Median_Con...</td>\n",
       "      <td>1.98%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GFKB v5 PreDiabetes</td>\n",
       "      <td>38 rows × 19 cols</td>\n",
       "      <td>Unique ID, UP ID, Present in GFKB v3 (Y/N), Pr...</td>\n",
       "      <td>14.27%</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Dataset               Shape  \\\n",
       "0     GFKB v4 Master List  188 rows × 18 cols   \n",
       "1  GFKB v4 Epilepsy Abund  188 rows × 43 cols   \n",
       "2     GFKB v5 PreDiabetes   38 rows × 19 cols   \n",
       "\n",
       "                                             Columns Missing % (total)  \n",
       "0  Unique ID, UP ID, Present in GFKB v3 (Y/N), Pr...             6.38%  \n",
       "1  uniqueID, UP name, Lineage( full ), Median_Con...             1.98%  \n",
       "2  Unique ID, UP ID, Present in GFKB v3 (Y/N), Pr...            14.27%  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def summarize_df(name, df, key_cols=5):\n",
    "    total_cells = df.size\n",
    "    missing_cells = df.isna().sum().sum()\n",
    "    missing_pct = round((missing_cells / total_cells) * 100, 2)\n",
    "\n",
    "    summary = {\n",
    "        \"Dataset\": name,\n",
    "        \"Shape\": f\"{df.shape[0]} rows × {df.shape[1]} cols\",\n",
    "        \"Columns\": \", \".join(df.columns[:key_cols]) + (\"...\" if len(df.columns) > key_cols else \"\"),\n",
    "        \"Missing % (total)\": f\"{missing_pct}%\"\n",
    "    }\n",
    "    return summary\n",
    "\n",
    "\n",
    "# Collect summaries\n",
    "summaries = [\n",
    "    summarize_df(\"GFKB v4 Master List\", df_v3),\n",
    "    summarize_df(\"GFKB v4 Epilepsy Abund\", df_v3_abund),\n",
    "    summarize_df(\"GFKB v5 PreDiabetes\", df_v5)\n",
    "]\n",
    "\n",
    "# Convert to DataFrame and display\n",
    "summary_df = pd.DataFrame(summaries)\n",
    "summary_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "83b820bb",
   "metadata": {},
   "source": [
    "### Prepare dataset for crossref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b4a84706",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Select only \"NCBI taxonomy name\" and add new columns\n",
    "filtered_df_v3 = df_v3[['Unique ID','UP name','NCBI taxonomy name','Assembly ID (UP matched)','Present in GFKB v3 (Y/N)','Present in GFKB_epilepsy v3 (Y/N)']].copy()\n",
    "filtered_df_v5 = df_v5[['Unique ID','UP name','NCBI taxonomy name','Assembly ID (UP matched)','Present in GFKB v3 (Y/N)','Present in GFKB_epilepsy v3 (Y/N)']].copy()\n",
    "\n",
    "# Prepare for concatenation\n",
    "filtered_df_v3['Present in GFKB v5 (Y/N)'] = 'N'\n",
    "filtered_df_v5['Present in GFKB v5 (Y/N)'] = 'Y'\n",
    "\n",
    "# Concatenation\n",
    "# Concatenate along rows (axis=0)\n",
    "result_df = pd.concat([filtered_df_v3, filtered_df_v5], ignore_index=True)\n",
    "\n",
    "# Add new columns (default 'N' and 'NA')\n",
    "result_df['In BiGG Models (Y/N)'] = 'Y/N'  # Default 'Y/N' (updated later in crossref)\n",
    "result_df['BiGG Model Link'] = None      # Default 'None' (can be updated later in crossref)\n",
    "\n",
    "# Sort by Name for easier crossreferecing\n",
    "result_df = result_df.sort_values(by=['UP name'])\n",
    "\n",
    "#Remove redundant rows\n",
    "result_df = result_df.drop_duplicates(subset=['UP name','NCBI taxonomy name','Present in GFKB v3 (Y/N)','Present in GFKB_epilepsy v3 (Y/N)']) #removes completely equal rows\n",
    "duplicates = result_df[result_df['UP name'].duplicated(keep=False)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "00a70d02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unique ID</th>\n",
       "      <th>UP name</th>\n",
       "      <th>NCBI taxonomy name</th>\n",
       "      <th>Assembly ID (UP matched)</th>\n",
       "      <th>Present in GFKB v3 (Y/N)</th>\n",
       "      <th>Present in GFKB_epilepsy v3 (Y/N)</th>\n",
       "      <th>Present in GFKB v5 (Y/N)</th>\n",
       "      <th>In BiGG Models (Y/N)</th>\n",
       "      <th>BiGG Model Link</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>UP000188159_CP012098</td>\n",
       "      <td>Anaerostipes hadrus</td>\n",
       "      <td>Anaerostipes hadrus</td>\n",
       "      <td>GCA_001998765.1</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>UP000008960_FP929061</td>\n",
       "      <td>Anaerostipes hadrus</td>\n",
       "      <td>Anaerostipes hadrus</td>\n",
       "      <td>GCA_000210695.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>UP000232402_CP011964</td>\n",
       "      <td>Bifidobacterium longum</td>\n",
       "      <td>Bifidobacterium longum subsp. longum</td>\n",
       "      <td>GCA_002833265.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>UP000232637_CP002286</td>\n",
       "      <td>Bifidobacterium longum</td>\n",
       "      <td>Bifidobacterium longum subsp. longum BBMN68</td>\n",
       "      <td>GCA_002832945.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>UP000031657_AP014658</td>\n",
       "      <td>Bifidobacterium longum</td>\n",
       "      <td>Bifidobacterium longum</td>\n",
       "      <td>GCA_000829295.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>UP000000439_AE014295</td>\n",
       "      <td>Bifidobacterium longum (strain NCC 2705)</td>\n",
       "      <td>Bifidobacterium longum NCC2705</td>\n",
       "      <td>GCA_000007525.1</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>UP000000439_LN824140</td>\n",
       "      <td>Bifidobacterium longum (strain NCC 2705)</td>\n",
       "      <td>Bifidobacterium longum subsp. infantis</td>\n",
       "      <td>GCA_000007525.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>UP000001377_CP021140</td>\n",
       "      <td>Eggerthella lenta (strain ATCC 25559 / DSM 224...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA_002148255.1</td>\n",
       "      <td>N</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>UP000001377_CP001726</td>\n",
       "      <td>Eggerthella lenta (strain ATCC 25559 / DSM 224...</td>\n",
       "      <td>Eggerthella lenta DSM 2243</td>\n",
       "      <td>GCA_000024265.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>UP000183509_CP006620</td>\n",
       "      <td>Enterococcus faecium (Streptococcus faecium)</td>\n",
       "      <td>Enterococcus faecium Aus0085</td>\n",
       "      <td>GCA_900080095.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>UP000190755_CP003351</td>\n",
       "      <td>Enterococcus faecium (Streptococcus faecium)</td>\n",
       "      <td>Enterococcus faecium Aus0004</td>\n",
       "      <td>GCA_002025045.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>UP000030306_CP009913</td>\n",
       "      <td>Streptococcus salivarius</td>\n",
       "      <td>Streptococcus salivarius</td>\n",
       "      <td>GCA_000785515.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>UP000234759_FR873482</td>\n",
       "      <td>Streptococcus salivarius</td>\n",
       "      <td>Streptococcus salivarius JIM8777</td>\n",
       "      <td>GCA_002860785.1</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>N</td>\n",
       "      <td>Y/N</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Unique ID                                            UP name  \\\n",
       "124  UP000188159_CP012098                                Anaerostipes hadrus   \n",
       "176  UP000008960_FP929061                                Anaerostipes hadrus   \n",
       "123  UP000232402_CP011964                             Bifidobacterium longum   \n",
       "53   UP000232637_CP002286                             Bifidobacterium longum   \n",
       "19   UP000031657_AP014658                             Bifidobacterium longum   \n",
       "1    UP000000439_AE014295           Bifidobacterium longum (strain NCC 2705)   \n",
       "186  UP000000439_LN824140           Bifidobacterium longum (strain NCC 2705)   \n",
       "136  UP000001377_CP021140  Eggerthella lenta (strain ATCC 25559 / DSM 224...   \n",
       "39   UP000001377_CP001726  Eggerthella lenta (strain ATCC 25559 / DSM 224...   \n",
       "92   UP000183509_CP006620       Enterococcus faecium (Streptococcus faecium)   \n",
       "77   UP000190755_CP003351       Enterococcus faecium (Streptococcus faecium)   \n",
       "114  UP000030306_CP009913                           Streptococcus salivarius   \n",
       "181  UP000234759_FR873482                           Streptococcus salivarius   \n",
       "\n",
       "                              NCBI taxonomy name Assembly ID (UP matched)  \\\n",
       "124                          Anaerostipes hadrus          GCA_001998765.1   \n",
       "176                          Anaerostipes hadrus          GCA_000210695.1   \n",
       "123         Bifidobacterium longum subsp. longum          GCA_002833265.1   \n",
       "53   Bifidobacterium longum subsp. longum BBMN68          GCA_002832945.1   \n",
       "19                        Bifidobacterium longum          GCA_000829295.1   \n",
       "1                 Bifidobacterium longum NCC2705          GCA_000007525.1   \n",
       "186       Bifidobacterium longum subsp. infantis          GCA_000007525.1   \n",
       "136                                          NaN          GCA_002148255.1   \n",
       "39                    Eggerthella lenta DSM 2243          GCA_000024265.1   \n",
       "92                  Enterococcus faecium Aus0085          GCA_900080095.1   \n",
       "77                  Enterococcus faecium Aus0004          GCA_002025045.1   \n",
       "114                     Streptococcus salivarius          GCA_000785515.1   \n",
       "181             Streptococcus salivarius JIM8777          GCA_002860785.1   \n",
       "\n",
       "    Present in GFKB v3 (Y/N) Present in GFKB_epilepsy v3 (Y/N)  \\\n",
       "124                        N                                 Y   \n",
       "176                        Y                                 Y   \n",
       "123                        Y                                 Y   \n",
       "53                         Y                                 Y   \n",
       "19                         Y                                 Y   \n",
       "1                          N                                 Y   \n",
       "186                        Y                                 Y   \n",
       "136                        N                                 Y   \n",
       "39                         Y                                 Y   \n",
       "92                         Y                                 Y   \n",
       "77                         Y                                 Y   \n",
       "114                        Y                                 Y   \n",
       "181                        Y                                 Y   \n",
       "\n",
       "    Present in GFKB v5 (Y/N) In BiGG Models (Y/N) BiGG Model Link  \n",
       "124                        N                  Y/N            None  \n",
       "176                        N                  Y/N            None  \n",
       "123                        N                  Y/N            None  \n",
       "53                         N                  Y/N            None  \n",
       "19                         N                  Y/N            None  \n",
       "1                          N                  Y/N            None  \n",
       "186                        N                  Y/N            None  \n",
       "136                        N                  Y/N            None  \n",
       "39                         N                  Y/N            None  \n",
       "92                         N                  Y/N            None  \n",
       "77                         N                  Y/N            None  \n",
       "114                        N                  Y/N            None  \n",
       "181                        N                  Y/N            None  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicates #This duplicates will cause a problem. Need to see assembly and understand differences between "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e6402ba3",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.to_csv('/home/arog/Documents/GitHub/HGMB_Project/datasets/treated/pre_crossref.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "global",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
